helix-view/src/editor.rs

```diff
@@ -74,6 +75,52 @@ where
     )
 }
 
+#[derive(Serialize, Deserialize, Default)]
+#[serde(rename_all = "kebab-case", default, deny_unknown_fields)]
+struct LayoutMap {
+    from: String,
+    into: String,
+}
+
+fn deserialize_layout_remap<'de, D>(deserializer: D) -> Result<HashMap<char, char>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    // Make (from, to) char pairs from `Vec` of `LayoutMap`s filtering
+    // cases where `from == to` which we don't need to translate.
+    let mut pairs = Vec::<LayoutMap>::deserialize(deserializer)?
+        .iter()
+        .flat_map(|layout_map| layout_map.from.chars().zip(layout_map.into.chars()))
+        .filter(|(from, to)| from != to)
+        .collect::<HashMap<char, char>>();
+    pairs
+        .iter()
+        .filter_map(|(_, to)| pairs.contains_key(to).then(|| *to))
+        .collect::<Vec<_>>()
+        .into_iter()
+        .for_each(|c| {
+            warn!(
+                "Key '{}' removed from layout-remap as it would overwrite Latin layout binding",
+                c
+            );
+            pairs.remove(&c);
+        });
```

Comment:
Is this part actually necessary? If you wanted to remap between keyboard layouts for example:

```toml
[[editor.layout-remap]]
# qwerty
from = "qwertyuiop[]asdfghjkl;'zxcvbnm,./"
# dvorak
into = "',.pyfgcrl/=aoeuidhtns-;qjkxbmwvz"
```

you would only rebind `[` and `]` since those are the only characters not duplicated between `from` and `into`. You only translate the incoming key event once so having duplication between `from` and `into` shouldn't matter - it won't cause some recursive loop of remapping.